src/Notification.php

```diff
@@ -56,18 +64,20 @@ public function isSuccess(): bool
 
     private static function validateNotificationData(array $data)
     {
-        $class_vars = get_class_vars(self::class);
-
         $missing = [];
         $invalid = [];
         $eventCodes = new \ReflectionClass(EventCodes::class);
 
-        foreach ($class_vars as $property => $value) {
+        foreach (self::REQUIRED_PROPERTIES as $property) {
+            // If required data is missing
             if (!isset($data[$property])) {
                 $missing[] = $property;
             }
-            if (isset($data['eventCode']) && !in_array($data['eventCode'], $eventCodes->getConstants())) {
-                $invalid[] = 'eventCode';
+
+            // If an invalid event code is passed
+            if (isset($data[self::PROPERTY_EVENT_CODE]) &&
+                !in_array($data[self::PROPERTY_EVENT_CODE], $eventCodes->getConstants())) {
+                $invalid[] = self::PROPERTY_EVENT_CODE;
```

Comment:
I see that when this `$invalid` array is used with the join below it doesn't have a separator. And the message refers to an invalid value, but wouldn't it be an invalid key in this context?

Reply:
Yes, the exception message wasn't very clear. Updated in 1007fa0

Reply:
@Morerice since this is a check for event codes, why not add the invalid event code itself into the `$invalid` array and let the error message be something like `Invalid event code`?
```suggestion
                $invalid[] = $data[self::PROPERTY_EVENT_CODE];
```
or maybe in order to accommodate validation for other fields in the future, you could set it up as an assoc array
```suggestion
                $invalid[self::PROPERTY_EVENT_CODE] = $data[self::PROPERTY_EVENT_CODE];
```

Reply:
Yes, I definitely agree with the associative array idea, but I think this is a bit out of scope of this PR, since it's goal was to add the `REFUND_FAILED` type. The addition of the constants was done just to ensure that the keys used are always identical.

A ticket to implement the assoc array will be created.